docs: add release notes for 0.29.0 (#8955)
determined-ai/determined: docs/release-notes.rst

 Release Notes
###############

**************
 Version 0.29
**************

Version 0.29.0
==============

**Release Date:** March 05, 2024

**Breaking Changes**

-  Add a new requirement for runtime configurations that there be a writable ``$HOME`` directory in
   every container. Previously, there was limited support for containers without a writable
   ``$HOME``, merely by coincidence. This change could impact users in scenarios where jobs were
   configured to run as the ``nobody`` user inside a container, instead of the ``det-nobody``
   alternative recommended in :ref:`run-unprivileged-tasks`. Users combining non-root tasks with
   custom images not based on Determined's official images may also be affected. Overall, it is
   expected that few or no users are affected by this change.

**Removed Features**

> -  Removed the accidentally-exposed Session object from the ``det.experimental.client`` namespace.

Reviewer: ```suggestion
-  Removed the accidentally exposed ``Session`` object from the ``det.experimental.client`` namespace.
```

   It was never meant to be a public API and it was not documented in :ref:`python-sdk`, but was
   nonetheless exposed in that namespace. It was also available as a deprecated legacy alias,
   ``det.experimental.Session``. It is expected that most users use the Python SDK normally and are
   unaffected by this change, since the ``det.experimental.client``'s ``login()`` and
   ``Determined()`` are unaffected.

**Improvements**

-  Configure log settings for the Determined agent in the configuration file used to launch

>    Determined clusters by setting ``log.level``and ``log.color`` appropriately.

Reviewer: ```suggestion
   Determined clusters by setting ``log.level`` and ``log.color`` appropriately.
```

**Bug Fixes**

> -  ResourceManager: Prevent connections from duplicate agents. Agent connection attempts will be

Reviewer: ```suggestion
-  Resource Manager: Prevent connections from duplicate agents. Agent connection attempts will be
```

   rejected if there's already an active connection from a matching agent ID. This prevents and

>    replaces previous behavior of stopping the running agent when a duplicate connection attempt is

Reviewer: ```suggestion
   replaces the previous behavior of stopping the running agent when a duplicate connection attempt is
```

   made (causing both connections to fail).

**Security**

-  Add a configuration setting, ``initial_user_password``, to the master configuration file forcing
   the setup of an initial user password for the built-in ``determined`` and ``admin`` users during

>    the first launch, specifically when a cluster's database is bootstrapped.

Reviewer: ```suggestion
   the first launch, when a cluster's database is bootstrapped.
```

.. important::

>    For any publicly-accessible cluster, you should ensure all users have a password set.

Reviewer: ```suggestion
   For any publicly accessible cluster, you should ensure all users have a password set.
```
